src/tribler/core/components/component.py

```diff
@@ -47,8 +50,16 @@ async def start(self):
             self._set_component_status(msg, logging.ERROR, exc_info=exc_info)
             self.failed = True
             self.started_event.set()
+
+            if isinstance(e, DatabaseIsCorrupted):
+                # When the database corruption is detected, we should stop the process immediately.
+                # Tribler GUI will restart the process and the database will be recreated.
+                process_manager = get_global_process_manager()
+                process_manager.sys_exit(EXITCODE_DATABASE_IS_CORRUPTED, e)
```

Comment:
Could be removed?

Reply:
Currently, it is necessary. The actual reason is a bit complex to explain:

An error that is raised during the component's startup is wrapped in `ComponentStartupException`. Then, it is handled in a special way:

```python
    def _reraise_startup_exception_in_separate_task(self):
        self.logger.info('Reraise startup exception in separate task')

        async def exception_reraiser():
            self.logger.info('Exception reraiser')

            e = self._startup_exception
            if isinstance(e, ComponentStartupException) and e.component.tribler_should_stop_on_component_error:
                self.logger.info('Shutdown with exit code 1')
                self.exit_code = 1
                self.shutdown_event.set()

            # the exception should be intercepted by event loop exception handler
            self.logger.info(f'Reraise startup exception: {self._startup_exception}')
            raise self._startup_exception

        self.async_group.add_task(exception_reraiser())
```

This logic triggers a weird asyncio behavior described in https://togithub.com/python/cpython/issues/69675. In short, if `CoreExceptionHandler.unhandled_error_observer` is called during the Tribler Core shutdown, the `SystemExit` error raised from `unhandled_error_observer` can be suppressed and ignored by asyncio. Apparently, `unhandled_error_observer` can be called inside the `Task.__del__()` method that ignores all re-raised exceptions, including the `SystemExit` error. As a result, Tribler Core finishes with exit code 1 (instead of code 99 dedicated to the database corruption error), and Tribler GUI does not restart the core.

Fixing and refactoring the logic of `unhandled_error_observer` can be a tricky and complicated task that is outside of the current PR scope. On the other side, raising `SystemExit` directly from the component's method works well and provides the desired results.

Also, raising `SystemExit` in the component allows TriblerCore to restart without waiting first until all components start.

Reply:
Could you please put this explanation into the code?